Allow adding a "lost password" page to the login module

zoglo opened the pull request:
**Feature**

- Implements #4710

**Description**

- This PR introduces the function of setting up a new redirect page for embedding the "Lost password" module in the login module in order to improve UX.

**Related**
- #6635
- #4710

___

It's a feature, not a bug - no need to merge it into 5.2

zoglo committed:
Add picker for lost-password redirect within login module (#4710)

zoglo committed:
Merge branch '5.x' into feature/login-lost-password

leofeyer commented:
I like the PR, however, it is kind of opinionated regarding the markup. @contao/developers What is the best position for the "reset password" link? Here are some examples:

<img width="336" alt="" src="https://github.com/contao/contao/assets/1192057/fa68e252-544f-4e22-802f-1488555efafc">

---

<img width="527" alt="" src="https://github.com/contao/contao/assets/1192057/7cfd24bd-6aeb-4ae8-8421-1f9e0874733f">

---

<img width="557" alt="" src="https://github.com/contao/contao/assets/1192057/bb732a01-72b0-4bcd-a028-447b1717546f">

---

<img width="556" alt="" src="https://github.com/contao/contao/assets/1192057/9ae4093f-711a-423b-9d60-2509537ca3b9">

ausi commented:
I think I’d put it directly after the `<div class="widget widget-submit">…</div>`

zoglo commented:
> I think I’d put it directly after the `<div class="widget widget-submit">…</div>`

Will change it right now. Makes the most sense to me as well.
If anyone needs it changed, they could use order with grid/flex.

zoglo committed:
Merge branch '5.x' into feature/login-lost-password

zoglo committed:
Move passwordReset-link within template

zoglo committed:
Merge branch '5.x' into feature/login-lost-password

zoglo committed:
Merge branch '5.x' into feature/login-lost-password

aschempp commented on core-bundle/contao/dca/tl_module.php:
```diff
@@ -73,7 +73,7 @@
 		'articlenav'                  => '{title_legend},name,headline,type;{config_legend},loadFirst;{template_legend:hide},customTpl;{protected_legend:hide},protected;{expert_legend:hide},cssID',
 		'sitemap'                     => '{title_legend},name,headline,type;{nav_legend},showProtected,showHidden;{reference_legend:hide},rootPage;{template_legend:hide},customTpl,navigationTpl;{protected_legend:hide},protected;{expert_legend:hide},cssID',
-		'login'                       => '{title_legend},name,headline,type;{config_legend},autologin;{redirect_legend},jumpTo,redirectBack;{template_legend:hide},customTpl;{protected_legend:hide},protected;{expert_legend:hide},cssID',
+		'login'                       => '{title_legend},name,headline,type;{config_legend},autologin;{redirect_legend},jumpTo,pwResetJumpTo,redirectBack;{template_legend:hide},customTpl;{protected_legend:hide},protected;{expert_legend:hide},cssID',
```
```suggestion
		'login'                       => '{title_legend},name,headline,type;{config_legend},autologin;{redirect_legend},jumpTo,redirectBack,pwResetJumpTo;{template_legend:hide},customTpl;{protected_legend:hide},protected;{expert_legend:hide},cssID',
```

The `redirectBack` checkbox has a relation to `jumpTo`, so they probably should stay together. Not sure what the UI will look like though.

zoglo replied:
It would look like this:
![image](https://github.com/contao/contao/assets/55794780/fa59f931-fde9-4e5e-b414-a2243842a901)

We could make use of w50 cbx m12 to have it next to the selection if you say that they should stay together:
![image](https://github.com/contao/contao/assets/55794780/f7fd8b4d-b9de-41c3-86d7-912f86809cb8)
![image](https://github.com/contao/contao/assets/55794780/e2dee426-863b-4248-a0df-d392858731d7)

Wdyt?

aschempp commented on core-bundle/contao/languages/de/default.xlf:
```diff
@@ -2325,6 +2325,10 @@
         <source>Quick link</source>
         <target>Quicklink</target>
       </trans-unit>
+      <trans-unit id="MSC.lostPassword">
```
German translations are managed on www.transifex.com, please don't add them to a pull request.

aschempp commented on core-bundle/contao/modules/ModuleLogin.php:
```diff
@@ -203,6 +203,15 @@ protected function compile()
+			$this->Template->passwordreset = $objLostPasswordTarget->getFrontendUrl();
+
+			$this->Template->resetTitle = $GLOBALS['TL_LANG']['MSC']['lostPassword'];
+			$this->Template->resetLabel = $GLOBALS['TL_LANG']['MSC']['lostPassword'];
```
any reason to have the same language variable in the template twice? Also, couldn't we just output that directly in the template? 
`<?= $this->trans('MSC.lostPassword', [], 'contao_default'); ?>` 

Not sure we're doing that anywhere, but I personally think the template should be responsible for the view data. @contao/developers any objections?

zoglo replied:
Shouldn't we change every translation within the LoginModule then?
Thus, we could remove the TL_LANG Superglobals one by one, however the translations within overridden templates won't exist anymore.

zoglo committed:
Merge branch '5.x' into feature/login-lost-password

zoglo committed:
Revert translations

zoglo committed:
Code cleanup

- Move lostPasswort field below redirectBack
- Revert translations
- Load translations within template

zoglo committed:
Update template

leofeyer committed:
Merge branch '5.x' into feature/login-lost-password

leofeyer committed:
CS

leofeyer committed:
Rename the `pwResetJumpTo` field to `pwResetPage`

leofeyer commented:
Thank you @zoglo.
